test/e2e_node/topology_manager_test.go

```diff
@@ -936,6 +940,90 @@ func runPreferClosestNUMATestSuite(ctx context.Context, f *framework.Framework,
 		podMap[podName] = pod
 	}
 
+	valiidatePreferClosestNUMAOptimalAllocation(ctx, f, podMap, numaNodes, distances)
+
+	deletePodsAsync(ctx, f, podMap)
```

Comment:
we likely will need to use `waitForContainerRemoval` to make sure the node state is fully restored. We got biten by this multiple time in the past

Reply:
this function inside already uses `waitForAllContainerRemoval`:
```go
func deletePodsAsync(ctx context.Context, f *framework.Framework, podMap map[string]*v1.Pod) {
	var wg sync.WaitGroup
	for _, pod := range podMap {
		wg.Add(1)
		go func(podNS, podName string) {
			defer ginkgo.GinkgoRecover()
			defer wg.Done()

			deletePodSyncByName(ctx, f, podName)
			waitForAllContainerRemoval(ctx, podName, podNS)
		}(pod.Namespace, pod.Name)
	}
	wg.Wait()
}
```

Reply:
cool, forgot about that (I probably even may have fixed this long ago!), thanks for checking